alchemy/src/cloudflare/website.ts

```diff
@@ -195,18 +190,41 @@ export default {
       } as WorkerProps<any> & { name: string };
 
       if (wrangler) {
-        // paths in wrangler.jsonc must be relative to it
-        const relativeToWrangler = <S extends string | undefined>(f: S): S =>
-          (f ? path.relative(path.dirname(wranglerJsonPath), f) : f) as S;
+        let inputWranglerPath =
+          typeof wrangler === "boolean"
+            ? undefined
+            : typeof wrangler === "string"
+              ? wrangler
+              : wrangler?.path;
+        let inputMainPath =
+          typeof wrangler === "object" && wrangler.main
+            ? wrangler.main
+            : props.main;
+        let inputAssetsDirPath = assetsDirPath;
+
+        if (inputWranglerPath && path.isAbsolute(inputWranglerPath)) {
+          const wranglerDir = path.dirname(wranglerJsonPath);
+          inputWranglerPath = path.relative(cwd, inputWranglerPath);
+          if (inputMainPath && path.isAbsolute(inputMainPath)) {
+            inputMainPath = path.relative(wranglerDir, inputMainPath);
+          }
+          inputAssetsDirPath = path.relative(wranglerDir, inputAssetsDirPath);
+        } else {
+          if (inputMainPath && path.isAbsolute(inputMainPath)) {
+            inputMainPath = path.relative(cwd, inputMainPath);
+          }
+          inputAssetsDirPath = path.relative(cwd, inputAssetsDirPath);
+        }
```

Comment:
I didn't expect this to need to change? Couldn't it just pass through `cwd` and have `WranglerJson` do the resolution?